### PR TITLE
Implemented GUI

### PR DESCRIPTION
**Pull request by gamecoder95**
- Implemented the GUI for the auction system and all the controls
- Integrated auction client with the GUI through the use of a callback
- Fixed the way the auction client was storing offers it made such that they are indexed by item number and not the request number.

At this point, the GUI is testable but not fully functional:
- If you run the test script, you can register with the GUI, but when attempting to send an offer, you offer denied saying that you're "not registered".

### CODE REVIEW

File: auctionsystem/client.py [hunk 1]
```diff
@@ -98,134 +104,124 @@ def handle_receive(self, data, addr=None):
...
-            print('resending register message')
-            self.send_register(name=msg_args[0], ip_addr=msg_args[1], port_num=msg_args[2],
-                               resending=True, req_num_resend=req_num)
+            print('Could not register because {}'.format(reason.str))
 
     def rcv_dereg_conf(self, req_num, name, ip_addr):
+        # Handle UDP message to confirm registration for the client
         self.confirm_acknowledgement(req_num)
-        self.tcp_clients.clear()  # Should close all of the connections
-        self.udp_client.close_socket()
```
Comment (samratdebroy) [inline auctionsystem/client.py]: I think these two lines should not have been removed, prolly got lost during merge conflict resolution

**Review by samratdebroy — CHANGES_REQUESTED** [review]
Gui looks good, just need to readd those two missing lines in auctionsystem client that we due to merge conflicts

File: auctionsystem/client.py (hunk 1 above)
Comment (gamecoder95) [reply auctionsystem/client.py]: Good catch! Added them back in.

**Review by samratdebroy — APPROVED** [review]
looks good

**Pull request merged**